Sprint 14 - ShareIt v.1

DmitryBrosalin opened the pull request:
Добавление, изменение, получение и удаление пользователей. Добавление, изменение, получение и поиск вещей.

DmitryBrosalin committed:
Sprint 14 - ShareIt v.1
Добавление, изменение, получение и удаление пользователей.
Добавление, изменение, получение и поиск вещей.

DmitryBrosalin committed:
Sprint 14 - ShareIt v.1
Добавление, изменение, получение и удаление пользователей.
Добавление, изменение, получение и поиск вещей.

slavrikk commented on src/main/java/ru/practicum/shareit/item/InMemoryItemRepository.java:
```diff
@@ -0,0 +1,73 @@
+
+    public ItemDto getItem(long id) {
+        try {
+            Optional<Item> itemOpt = Optional.of(items.get(id));
```
Преде вызовам метода get можно проверить на наличие `item` в коллекции, затем выбросить ошибку

slavrikk commented on src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -1,12 +1,43 @@
+    private final ItemService itemService;
+
+    @PostMapping
+    public ItemDto addItem(@RequestHeader("X-Sharer-User-Id") long userId, @RequestBody Item item) {
```
В контроллер должны приходить так же `ItemDto`

slavrikk replied:
Замечание для всех методов

slavrikk commented on src/main/java/ru/practicum/shareit/user/InMemoryUserRepository.java:
```diff
@@ -0,0 +1,66 @@
+
+    public User getUser(long id) {
+        try {
+            Optional<User> userOpt = Optional.of(users.get(id));
```
Аналогичный комментарий

slavrikk commented on src/main/java/ru/practicum/shareit/user/InMemoryUserRepository.java:
```diff
@@ -0,0 +1,66 @@
+
+    public User updateUser(long id, User user) {
+        User oldUser = getUser(id);
+        users.remove(id);
```
Можно не удалять объект, достаточно заменить данные в нем же

slavrikk commented on src/main/java/ru/practicum/shareit/user/UserController.java:
```diff
@@ -1,12 +1,32 @@
+    private final UserService userService;
+
+    @PostMapping
+    public User addUser(@Valid @RequestBody User user) {
```
Тут должен передаваться `UserDto` и возвращаться `UserDto`

slavrikk replied:
Для остальных методов аналогично

DmitryBrosalin committed:
Sprint 14 - ShareIt v.2
Изменил логику проверки наличия объектов в коллекциях перед вызовом get-методов.
Заменил классы, с которыми работают контроллеры, на dto.

DmitryBrosalin committed:
Sprint 14 - ShareIt v.2
Изменил логику проверки наличия объектов в коллекциях перед вызовом get-методов.
Заменил классы, с которыми работают контроллеры, на dto.
